packages/core/src/request-builder/header-builder/headers-util.ts

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
+import { MapType } from '@sap-cloud-sdk/util';
+
+export function isNullish(x: any): x is null | undefined {
+  return x === null || x === undefined;
+}
+
+export function toHeaderObject(key: string, value: any): MapType<any> {
+  return isNullish(value) ? {} : { [key]: value };
+}
+
+export function getHeader(
+  key: string,
+  headers: MapType<any> = {}
+): MapType<any> {
+  const entry = Object.entries(headers).find(
+    ([entryKey]) => entryKey.toLowerCase() === key.toLowerCase()
+  );
+  return entry ? { [entry[0]]: entry[1] } : {};
+}
+
+export function getHeaderValue(
+  key: string,
+  headers: MapType<any> = {}
+): any | undefined {
+  return Object.entries(headers).find(
+    ([entryKey]) => entryKey.toLowerCase() === key.toLowerCase()
+  )?.[1];
+}
+
+export function filterNullishValues(headers: MapType<any> = {}): MapType<any> {
+  return Object.entries(headers)
+    .filter(([_, value]) => !isNullish(value))
+    .reduce((filtered, [key, value]) => ({ ...filtered, [key]: value }), {});
```

Comment:
what is the reduce doing after the filter? Could you not just return the filtered? And why do you use `_` as name for the key in the filter?

Reply:
The secret here is: Object.entries(headers).
1. The reduce transforms the array of entries [key, value][] to an object.
2. no, because of 1.
3. I don't need the key here and want to make it clear by basically ignoring the value (I need it in the reduce though)